File: ListeDeGarde/Varia/Ribbon1.vb
Imports Microsoft.Office.Tools.Ribbon

Public Class Ribbon1
    Private WithEvents aform1 As Form1
    Private Sub Ribbon1_Load(ByVal sender As System.Object, ByVal e As RibbonUIEventArgs) Handles MyBase.Load

    End Sub

    Private Sub Button1_Click(sender As Object, e As RibbonControlEventArgs) Handles Button1.Click
        ' UseWithEvents()
        Globals.ThisAddIn.taskpane.visible = True
    End Sub

    Private Sub Button2_Click(sender As Object, e As RibbonControlEventArgs) Handles Button2.Click
        'UseDelegate()
        aform1 = New Form1
        aform1.ShowDialog()
    End Sub

    Private Sub Button3_Click(sender As Object, e As RibbonControlEventArgs) Handles Button3.Click
        LoadDatabaseFileLocation()
    End Sub

    Private Sub Button4_Click(sender As Object, e As RibbonControlEventArgs) Handles Button4.Click
        Dim aform1 As New DrInterfaceForm
        aform1.ShowDialog()
    End Sub

    Private Sub ShiftButton_Click(sender As Object, e As RibbonControlEventArgs) Handles ShiftButton.Click
        Dim aform1 As New ShiftInterfaceF
        aform1.ShowDialog()
    End Sub

    Private Sub aForm1_close() Handles aform1.FormClosing
        If Not Globals.ThisAddIn.theControllerCollection.Contains(Globals.ThisAddIn.Application.ActiveSheet.name) Then Exit Sub
        Dim aController As Controller = Globals.ThisAddIn.theControllerCollection.Item(Globals.ThisAddIn.Application.ActiveSheet.name)
        aController.resetSheetExt()
    End Sub

 
    Private Sub ExpectDoc_Click(sender As Object, e As RibbonControlEventArgs) Handles ExpectDoc.Click
        Dim theController As Controller
        If Globals.ThisAddIn.theControllerCollection.Count < 1 Then Exit Sub
        If Not Globals.ThisAddIn.theControllerCollection.Contains(Globals.ThisAddIn.Application.ActiveSheet.name) Then Exit Sub
        theController = Globals.ThisAddIn.theControllerCollection(Globals.ThisAddIn.Application.ActiveSheet.name)

        Dim aDocExpecationF As DocExpectationsF
        aDocExpecationF = New DocExpectationsF
        aDocExpecationF.Show()
    End Sub
End Class

File: ListeDeGarde/CS/Varia/Ribbon1.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using Office = Microsoft.Office.Core;
using Microsoft.VisualBasic;
using System.Xml.Linq;
using Microsoft.Office.Tools.Excel;
using Excel = Microsoft.Office.Interop.Excel;
using System.Text;
using System.Linq;
// End of VB project level imports

using Microsoft.Office.Tools.Ribbon;


namespace ListeDeGarde
{
	public partial class Ribbon1
	{
		private Form1 aform1;
		public void Ribbon1_Load(System.Object sender, RibbonUIEventArgs e)
		{
			
		}
		
		public void Button1_Click(object sender, RibbonControlEventArgs e)
		{
			// UseWithEvents()
			Globals.ThisAddIn.taskpane.Visible = true;
		}
		
		public void Button2_Click(object sender, RibbonControlEventArgs e)
		{
			//UseDelegate()
			aform1 = new Form1();
			aform1.FormClosing += new System.Windows.Forms.FormClosingEventHandler(aForm1_close);
			aform1.ShowDialog();
		}
		
		public void Button3_Click(object sender, RibbonControlEventArgs e)
		{
			GlobalFunctions.LoadDatabaseFileLocation();
		}
		
		public void Button4_Click(object sender, RibbonControlEventArgs e)
		{
			DrInterfaceForm aform1 = new DrInterfaceForm();
			aform1.ShowDialog();
		}
		
		public void ShiftButton_Click(object sender, RibbonControlEventArgs e)
		{
			ShiftInterfaceF aform1 = new ShiftInterfaceF();
			aform1.ShowDialog();
		}
		
		private void aForm1_close(System.Object sender, System.Windows.Forms.FormClosingEventArgs e)
		{
			if (!Globals.ThisAddIn.theControllerCollection.Contains((string) Globals.ThisAddIn.Application.ActiveSheet.name))
			{
				return;
			}
			Controller aController = Globals.ThisAddIn.theControllerCollection[Globals.ThisAddIn.Application.ActiveSheet.name];
			aController.resetSheetExt();
		}
		
		
		public void ExpectDoc_Click(object sender, RibbonControlEventArgs e)
		{
			Controller theController;
			if (Globals.ThisAddIn.theControllerCollection.Count < 1)
			{
				return;
			}
			if (!Globals.ThisAddIn.theControllerCollection.Contains((string) Globals.ThisAddIn.Application.ActiveSheet.name))
			{
				return;
			}
			theController = Globals.ThisAddIn.theControllerCollection[Globals.ThisAddIn.Application.ActiveSheet.name];
			
			DocExpectationsF aDocExpecationF = default(DocExpectationsF);
			aDocExpecationF = new DocExpectationsF();
			aDocExpecationF.Show();
		}
	}
	
}
